test/model/test/common/wiredtiger_util.cpp

```diff
@@ -319,6 +322,99 @@ wt_txn_insert(WT_SESSION *session, const char *uri, const model::data_value &key
     return ret;
 }
 
+/*
+ * wt_ckpt_get --
+ *     Read from WiredTiger.
+ */
+model::data_value
+wt_ckpt_get(WT_SESSION *session, const char *uri, const model::data_value &key,
+  const char *ckpt_name, model::timestamp_t debug_read_timestamp)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    size_t config_len;
+    char buf[64];
+    char *config;
+    model::data_value out;
```

Comment:
This function reads a bit like it was ported from C - I think there's some cleanup opportunities here, e.g. declaring variables where they're used, and using a `stringstream` instead of the `alloca`/`printf` combo. (Even if you don't want to use a `stringstream`, I would prefer to get rid of the `alloca`, here and below - we don't know the length of `ckpt_name` ahead of time and I've been bitten before by code like this.)

Reply:
I forget whether this uses C++17 or C++20 - if the latter, `std::format` becomes an option.

Reply:
Ok, done—I refactored it using `ostringstream`. `std::format` would be neater, but it requires C++20.